pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -151,9 +151,9 @@ func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s ku
 	switch obj := obj.(type) {
 	case *v1beta1.Pipeline:
 		// Verify the Pipeline once we fetch from the remote resolution, mutating, validation and conversion of the pipeline should happen after the verification, since signatures are based on the remote pipeline contents
-		err := trustedresources.VerifyPipeline(ctx, obj, k8s, refSource, verificationPolicies)
-		if err != nil {
-			return nil, fmt.Errorf("remote Pipeline verification failed for object %s: %w", obj.GetName(), err)
+		vr := trustedresources.VerifyPipeline(ctx, obj, k8s, refSource, verificationPolicies)
+		if vr.VerificationResultType == trustedresources.VerificationError && vr.Err != nil {
+			return nil, fmt.Errorf("remote Pipeline verification failed for object %s: %w", obj.GetName(), vr.Err)
```

Comment:
Those lines of code are temporary changes, will be updated in next PR

Reply:
I think if the result type is VerificationError, that would signify an error even if the message is nil, and we would just have to decide what error to return. Are there any cases where result type could be error, with vr.Err = nil?

Reply:
checking if it's nil when creating the error message is fine, the point I was trying to make was just more that if vr has a VerificationError, we should probably still be returning an error rather than nil

Reply:
Oh I see. It would be something like this?
```
if vr.VerificationResultType == trustedresources.VerificationError {
if  vr.Err != nil{
return vr.Err
}else{
return fmt.Error("....")
}
}
```
I just feel like there's no need to change it, we will return the vr directly in next PR

Reply:
right now if error is nil but vr.VerificationResultType is trustedresources.VerificationError, we treat this as if there's no error, that doesn't seem right to me.
I would do
```
if vr.VerificationResultType == trustedresources.VerificationError {
   if err != nil { /* same as what exists */ }
   return nil, fmt.Errorf("remote Pipeline verification failed for object %s", obj.GetName())
}
```

Reply:
ok

Reply:
updated